sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -454,6 +481,56 @@ export class ServiceBusReceiverImpl implements ServiceBusReceiver {
     return retry<ServiceBusReceivedMessage[]>(config);
   }
 
+  async deleteMessages(options: DeleteMessagesOptions): Promise<number> {
+    this._throwIfReceiverOrConnectionClosed();
+
+    const deleteMessagesOperationPromise = (): Promise<number> => {
+      return this._context
+        .getManagementClient(this.entityPath)
+        .deleteMessages(options.maxMessageCount, options?.beforeEnqueueTime, undefined, {
+          ...options,
+          associatedLinkName: this._getAssociatedReceiverName(),
+          requestName: "deleteMessages",
+          timeoutInMs: this._retryOptions.timeoutInMs,
+        });
+    };
+    const config: RetryConfig<number> = {
+      operation: deleteMessagesOperationPromise,
+      connectionId: this._context.connectionId,
+      operationType: RetryOperationType.management,
+      retryOptions: this._retryOptions,
+      abortSignal: options?.abortSignal,
+    };
+    return retry<number>(config);
+  }
+
+  async purgeMessages(options?: PurgeMessagesOptions): Promise<number> {
+    let deletedCount = await this.deleteMessages({
+      maxMessageCount: MaxDeleteMessageCount,
+      beforeEnqueueTime: options?.beforeEnqueueTime,
+    });
+    logger.verbose(
+      `${this.logPrefix} receiver '${this.identifier}' deleted ${deletedCount} messages.`,
+    );
+    if (deletedCount > 0) {
```

Comment:
revert to the same logic as in `sessionReceiver` once service fix issue to ensure deleting `MaxDeleteMessageCount` messages.